cmd/server/main.go

```
@@ -0,0 +1,198 @@
+		}
+		resp.Payload = mustMarshal(commands.PutDocumentResponse{})
+
+	case commands.GetDocumentCommand:
+		var req commands.GetDocumentRequest
+		if err := json.Unmarshal(in.Payload, &req); err != nil {
+			return resp, err
+		}
+		coll, err := store.GetCollection(req.Collection)
+		if err != nil {
+			return resp, err
+		}
+		doc, err := coll.Get(req.Key)
+		if err != nil {
+			r := commands.GetDocumentResponse{Found: false}
+			resp.Payload = mustMarshal(r)
+			return resp, nil
+		}
+
+		m := make(map[string]interface{})
+		if err := documentstore.UnmarshalDocument(doc, &m); err != nil {
+			return resp, err
+		}
+		r := commands.GetDocumentResponse{
+			Document: m,
+			Found:    true,
+		}
+		resp.Payload = mustMarshal(r)
+
+	case commands.DeleteDocumentCommand:
+		var req commands.DeleteDocumentRequest
+		if err := json.Unmarshal(in.Payload, &req); err != nil {
+			return resp, err
+		}
+		coll, err := store.GetCollection(req.Collection)
+		if err != nil {
+			return resp, err
+		}
+		success := coll.Delete(req.Key) == nil
+		r := commands.DeleteDocumentResponse{Success: success}
+		resp.Payload = mustMarshal(r)
+
+	case commands.ListDocumentsCommand:
+		var req commands.ListDocumentsRequest
+		if err := json.Unmarshal(in.Payload, &req); err != nil {
+			return resp, err
+		}
+		coll, err := store.GetCollection(req.Collection)
+		if err != nil {
+			return resp, err
+		}
+		docs := coll.List()
+
+		outDocs := make([]map[string]interface{}, 0, len(docs))
+		for _, d := range docs {
+			m := make(map[string]interface{})
+			if err := documentstore.UnmarshalDocument(&d, &m); err != nil {
+				return resp, err
+			}
+			outDocs = append(outDocs, m)
+		}
+		r := commands.ListDocumentsResponse{Documents: outDocs}
+		resp.Payload = mustMarshal(r)
+
+	default:
+		resp.Payload, _ = json.Marshal(map[string]string{"error": "unknown command"})
+	}
+
+	return resp, nil
+}
+
+func sendMessage(w *bufio.Writer, msg commands.CommandMessage) {
+	data, _ := json.Marshal(msg)
```

Silently ignoring errors is not a good strategy